src/Compilers/CSharp/Test/Semantic/Semantics/NonTrailingNamedArgumentsTests.cs

```diff
@@ -0,0 +1,646 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.Semantics
+{
+    [CompilerTrait(CompilerFeature.NonTrailingNamedArgs)]
+    public class NonTrailingNamedArgumentsTests : CompilingTestBase
+    {
+        [Fact]
+        public void TestSimple()
+        {
+            var source = @"
+class C
+{
+    static void M(int a, int b)
+    {
+        System.Console.Write($""First {a} {b}. "");
+    }
+    static void M(long b, long a)
+    {
+        System.Console.Write($""Second {b} {a}. "");
+    }
+    static void Main()
+    {
+        M(a: 1, 2);
+        M(3, a: 4);
+    }
+}";
+            var verifier = CompileAndVerify(source, expectedOutput: "First 1 2. Second 3 4.", parseOptions: TestOptions.Regular7_2);
```

Comment:
I kept one test to use 7.2 specifically to demonstrate that the feature is available starting from 7.2.